pebblo/app/config/config_validation.py

```diff
@@ -0,0 +1,72 @@
+import os
+from pebblo.app.utils.utils import get_full_path
+from pebblo.app.libs.logger import logger
+from abc import ABC, abstractmethod
+import sys
+import logging
+
+
+class ConfigValidator(ABC):
+    def __init__(self, config):
+        self.config = config
+        self.errors = []
+
+    @abstractmethod
+    def validate(self):
+        pass
+
+
+class DaemonConfig(ConfigValidator):
+    def validate(self):
+        host = self.config.get("host")
+        port = self.config.get("port")
+
+        if not isinstance(host, str):
+            self.errors.append("Error: Host must be a string")
+        try:
+            port = int(port)
+            if not (0 < port <= 65535):
+                self.errors.append("Error: Port must be between 1 and 65535")
+        except ValueError:
+            self.errors.append("Error: Port must be an integer")
+
+
+class LoggingConfig(ConfigValidator):
+    def validate(self):
+        level = self.config.get("level", "").upper()
+        if level not in logging._nameToLevel:
+            self.errors.append("Error: Unsupported logLevel specified in the configuration")
+
+
+class ReportsConfig(ConfigValidator):
+    def validate(self):
+        format_ = self.config.get("format")
+        renderer = self.config.get("renderer")
+        output_dir = self.config.get("outputDir")
+
+        if format_ not in ["pdf"]:
+            self.errors.append("Error: Unsupported format specified in the configuration")
+        if renderer not in ["weasyprint", "xhtml2pdf"]:
+            self.errors.append("Error: Unsupported renderer specified in the configuration")
```

Comment:
Nit pick : Please add wrong value in the message so that it's clear to the user.

Reply:
Revised the code to reflect the above improvements.